src/services/stats-wales-api.ts

```diff
@@ -149,6 +169,38 @@ export class StatsWalesApi {
         }).then((response) => response.body as ReadableStream);
     }
 
+    public async getRevisionCubeCSV(datasetId: string, revisionId: string): Promise<ReadableStream> {
+        logger.debug(`Fetching CSV for revision: ${revisionId}...`);
+
+        return this.fetch({
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/cube/csv`
+        }).then((response) => response.body as ReadableStream);
+    }
+
+    public async getRevisionCubeParquet(datasetId: string, revisionId: string): Promise<ReadableStream> {
+        logger.debug(`Fetching CSV for revision: ${revisionId}...`);
+
+        return this.fetch({
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/cube/parquet`
+        }).then((response) => response.body as ReadableStream);
+    }
+
+    public async getRevisionCubeExcel(datasetId: string, revisionId: string): Promise<ReadableStream> {
+        logger.debug(`Fetching CSV for revision: ${revisionId}...`);
+
+        return this.fetch({
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/cube/excel`
+        }).then((response) => response.body as ReadableStream);
+    }
+
+    public async getRevisionCube(datasetId: string, revisionId: string): Promise<ReadableStream> {
+        logger.debug(`Fetching CSV for revision: ${revisionId}...`);
+
+        return this.fetch({
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/cube`
+        }).then((response) => response.body as ReadableStream);
+    }
```

Comment:
As above, this could probably be 

```ts
public async getRevisionCube(datasetId: string, revisionId: string, format: FileFormat): Promise<ReadableStream> {
    logger.debug(`Fetching CSV for revision: ${revisionId}...`);

    return this.fetch({
        url: `dataset/${datasetId}/revision/by-id/${revisionId}/cube?format=${format}`
    }).then((response) => response.body as ReadableStream);
}
```